### PR TITLE
[Lang] Add get_addr() to retrieve the buffer addr of an SNode

### PR DESCRIPTION
**Pull request by AmesingFlank**
Add get_addr() to retrieve the pointer of an SNode. This only works on LLVM backends for now.

### CODE REVIEW

**Review by k-ye — COMMENTED** [review]
I think we need to document this PR more clearly -- Originally I thought that `get_addr` was intended to be used on the host side. Also there seems to be a few unrelated examples, such as `demo_sparse*`, `mpm88_sparse`, `performance_cacheline`.

File: examples/print_offset.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import taichi as ti
+
+ti.init(arch=ti.cpu, print_ir=True)
+
+n = 4
+m = 8
+
+# a = ti.field(dtype=ti.u8, shape=N)
+a = ti.field(dtype=ti.i32)
+# ti.root.dense(ti.ij, (4, 8)).place(a)
```
Comment (k-ye) [inline examples/print_offset.py:10]: Clean up?

Comment (AmesingFlank) [reply examples/print_offset.py:10]: Sure!

**Comment by AmesingFlank** [comment]
> I think we need to document this PR more clearly -- Originally I thought that `get_addr` was intended to be used on the host side. Also there seems to be a few unrelated examples, such as `demo_sparse*`, `mpm88_sparse`, `performance_cacheline`.

Sure. I'll remove add some comments and remove the unrelated examples from this PR.

File: python/taichi/lang/ops.py [hunk 2]
```diff
@@ -579,3 +579,22 @@ def rescale_index(a, b, I):
         if a.shape[n] < b.shape[n]:
             Ib.entries[n] = I.entries[n] * (b.shape[n] // a.shape[n])
     return Ib
+
+
+def get_addr(f, indices):
```
Comment (k-ye) [inline python/taichi/lang/ops.py:584]: Sorry my bad, we actually follow the Google docstring style... See https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Comment (AmesingFlank) [reply python/taichi/lang/ops.py:584]: Understood. will fix.

File: python/taichi/lang/ops.py [hunk 3]
```diff
@@ -579,3 +579,17 @@ def rescale_index(a, b, I):
...
+    """Returns the address (for CUDA/x64) for index 'indices' of field 'f'. Currently, this function can only be called inside a kernel.
```
Comment (k-ye) [inline python/taichi/lang/ops.py:585]: super nits: indices and f should be quoted with a single backquote "`"

**Review by k-ye — APPROVED** [review]
Awesome LGTM!

File: python/taichi/lang/ops.py (hunk 3 above)
Comment (AmesingFlank) [reply python/taichi/lang/ops.py:585]: Ah right. will fix.

**Pull request merged**